src/psyclone/transformations.py

```diff
@@ -405,6 +405,40 @@ def name(self):
         ''' Returns the name of this transformation as a string.'''
         return "GOceanLoopFuse"
 
+    def validate(self, node1, node2, options=None):
+        '''Checks if it is valid to apply the GOceanLoopFuseTrans
+        transform. It ensures that the fused loops are over
+        the same grid-point types, before calling the normal
+        LoopFuseTrans validation function.
+
+        :param node1: the first Node representing a GOLoop.
+        :type node1: :py:class:`psyclone.gocean1p0.GOLoop`
+        :param node2: the second Node representing a GOLoop.
+        :type node2: :py:class:`psyclone.gocean1p0.GOLoop`
+        :param options: a dictionary with options for transformations.
+        :type options: dictionary of string:values or None
+
+        :raises TransformationError: if the supplied loops are over \
+                                     different grid-point types.
+
+        :raises TransformationError: if invalid parameters are passed in.
+
+        '''
+
+        try:
+            if node1.field_space != node2.field_space:
+                raise TransformationError(
+                    "Error in {0} transformation. Cannot "
+                    "fuse loops that are over different grid-point types: "
+                    "{1} {2}".format(self.name, node1.field_space,
+                                     node2.field_space))
+        except Exception as err:
```

Comment:
Ha, that got me as well. The test case provides a omp-directive or so as 'node', which then raises a 'no such member' (or so) error in 'node1.field_space != node2.field_space'.

Now ... I could have moved the call to validate of the base class to be done first in this validate function ... it tests if the nodes are indeed Loop instances - so the test would work. BUT: what if we provide a Dynamo-loop? Then the base class validate would still identify the nodes as Loops, but they would not have a field_space ... and the code would crash.

So I re-added the try-loop (we could alternatively test if the Loops are gocean loops, but considering that we had the same code in the past, I thought to just leave it this way). Let me know which way you want to go :)

Reply:
Since this is now a GOcean-specific validate, I think we should/could add a check that the two nodes are instances of GOLoop before checking the spaces.

Reply:
I've added a GOLoop specific tests (and improved some tests to check for the right error message).

Reply:
Thanks.